### DISCUSSION TITLE
Custom variant not working

### DISCUSSION THREAD

**Discussion by mattiaz9** [tailwindlabs/tailwindcss#5344] [Help]
I'm using tailwind with next.js and jit mode enabled. 
When I try to use a custom variant I either get this error: `Syntax error: Selector "" is not pure (pure selectors must contain at least one local class or id)`, or if don't get an error the utility is just not added to the selector.

I tried with the example provided on the website:
```
addVariant("important", ({ container }) => {
  container.walkRules(rule => {
    rule.selector = `.\\!${rule.selector.slice(1)}`
    rule.walkDecls(decl => {
       decl.important = true
    })
  })
})
```

in my scss:
```
.heading {
  @apply important:text-red-400;
}
```

I'm not sure if it's a bug or I'm doing something wrong.

**Comment by losnappas** [comment]
Fyi, that variant is already a default (with jit): `!text-red-400`. https://tailwindcss.com/docs/just-in-time-mode#built-in-important-modifier
